client/state/themes/upload-theme/reducer.js

```diff
@@ -247,11 +247,45 @@ export const isTransferComplete = ( state = {}, action ) => {
 	return state;
 };
 
+export const isTransferInProgress = ( state = {}, action ) => {
+	switch ( action.type ) {
+		case THEME_TRANSFER_INITIATE_REQUEST:
+		case THEME_TRANSFER_INITIATE_PROGRESS: {
+			const { siteId } = action;
+
+			return {
+				...state,
+				[ siteId ]: true,
+			};
+		}
+		case THEME_TRANSFER_INITIATE_FAILURE:
+		case THEME_TRANSFER_STATUS_FAILURE: {
+			const { siteId } = action;
+
+			return {
+				...state,
+				[ siteId ]: false,
+			};
+		}
+		case THEME_TRANSFER_STATUS_RECEIVE: {
+			const { siteId, status } = action;
+
+			return {
+				...state,
+				[ siteId ]: status !== 'complete',
+			};
+		}
+	}
+
+	return state;
+};
```

Comment:
> inProgress also contains the events for the theme upload

Sorry, I don't follow. I wasn't asking why we cannot re-use the `inProgress` prop.

I was asking why we add a new reducer and a new selector that checks whether the transfer is active, when we already have existing utils like `isAutomatedTransferActive`.

Reply:
Related: I see the component has a `isTransferring` state. Cannot we use that as well?

Reply:
>Sorry, I don't follow. I wasn't asking why we cannot re-use the inProgress prop.

Oops, I should probably stop responding to PR comments at 9 pm 😅 

> I was asking why we add a new reducer and a new selector that checks whether the transfer is in progress, when we already have existing utils like isAutomatedTransferActive.

The ThemeUpload component uses `initiateThemeTransfer` that dispatches its own transfer status events (e.g. `THEME_TRANSFER_INITIATE_REQUEST`) and stores them in a separate state. I'm not sure if `isAutomatedTransferActive` would work, but even if it works, it would mean having the status extracted from multiple places. 

> Related: I see the component has a isTransferring state. Cannot we use that as well?

That prop seems to be populated from `HostingActivateStatus` component; which for some reason doesn't seem to work nice with the Theme Upload implementation.

Reply:
> `initiateThemeTransfer` that dispatches its own transfer status events

Seems that the `automated-transfer` state use the same actions:

https://github.com/Automattic/wp-calypso/blob/4b4c0f9e7c12d7e639a8ef06ffd1e4648c75ef04/client/state/automated-transfer/reducer.js#L9-L11

I have the feeling we're duplicating the code here.

Reply:
Yeah, it's duplicated. I'll give it a try to re-implement it to use the generic AT logic. I didn't want to change the current architecture, but it would probably make the code more cleaner.

Reply:
I've tried using that selector instead but the implementation is not working as expected: the notice ui appears for a second and dissapears and re-appears after 20s. This is because the status is [refetched](https://github.com/Automattic/wp-calypso/blob/4b55037b44daee8daacd0f4b49937f687e0b62a7/client/my-sites/hosting/hosting-activate-status.tsx#L69) immediately, before the AT transfer actually starts:
- initiate request is made (request pending) -> AT Status: start
- the notice appears
- An AT fetch status is requested - the AT process hasn't started -> new status none
- The notice dissapears
- initiate request is finished (usually takes 10-15-20 seconds) - it's a file upload after all
- A fetch status is requested -> new status is now provisioning
- The notice appears again

We don't have this problem with the theme AT state because it behaves differently, it's not using only the AT revert states to determine if the AT is in progress. 

Implementing a proper solution in the automated-transfer state would require a bit more effort and I'm not 100% sure we won't introduce regressions.